pkg/catalog/dispatcher.go

```diff
@@ -0,0 +1,108 @@
+package catalog
+
+import (
+	"reflect"
+	"strings"
+	"time"
+
+	a "github.com/openservicemesh/osm/pkg/announcements"
+	"github.com/openservicemesh/osm/pkg/kubernetes/events"
+)
+
+const (
+	maxBroadcastDeadlineTime = 15 * time.Second
+	maxGraceDeadlineTime     = 3 * time.Second
+)
+
+func (mc *MeshCatalog) dispatcher() {
+	// This will be finely tunned in near future, we can instrument other modules
+	// to take ownership of certain events, and just notify dispatcher through
+	// ScheduleBroadcastUpdate announcement type
+	subChannel := events.GetPubSubInstance().Subscribe(
```

Comment:
I wonder if there's a way to use the `reflect` package and get all enums.

I found this hack, which seems reasonable: https://play.golang.org/p/azPEcyTw6RQ -- to ensure that we automatically add values here to this list.

Reply:
I'm not getting all of them..., plus I suspect we might move some of them in very near future to be instead individually handled elsewhere with some more specific logic.